apps/data/database/initdb.sql

```diff
@@ -0,0 +1,68 @@
+CREATE TYPE STATUS_TYPE AS ENUM ('AVAILABLE', 'SOLD');
+
+CREATE TYPE LOCATION_TYPE AS (
+    latitude float,
+    longitude float
+);
+
+-- Creating Tables:
+
+CREATE TABLE UserTable (
+    user_id SERIAL PRIMARY KEY,
+    username VARCHAR NOT NULL,
+    email VARCHAR NOT NULL,
+    password VARCHAR NOT NULL,
+    name VARCHAR,
+    bio TEXT,
+    profile_pic_url VARCHAR,
+    joining_date TIMESTAMP DEFAULT NOW(),
+    listings_sold INTEGER[],
+    listings_purchased INTEGER[]
+);
+
+CREATE TABLE ListingTable (
+    listing_id SERIAL PRIMARY KEY,
+    seller_id INTEGER REFERENCES UserTable(user_id),
+    title VARCHAR NOT NULL,
+    description VARCHAR NOT NULL,
+    price INTEGER NOT NULL,
+    location LOCATION_TYPE,
+    status STATUS_TYPE,
+    date_created TIMESTAMP DEFAULT NOW(),
+    date_modified TIMESTAMP,
+    image_urls TEXT[]
+);
+
+CREATE TABLE MessageTable (
+    message_id SERIAL PRIMARY KEY,
+    sender_id INTEGER REFERENCES UserTable(user_id),
+    receiver_id INTEGER REFERENCES UserTable(user_id),
+    listing_id INTEGER REFERENCES ListingTable(listing_id),
+    message_body TEXT NOT NULL,
+    date_created TIMESTAMP DEFAULT NOW()
+);
+
+CREATE TABLE RatingTable (
```

Comment:
I think we are going with Reviews with a rating attached rather than Ratings with a review attached for naming